cumulus_logger.py

```diff
@@ -1,99 +1,315 @@
+
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
+        self._log(logging.DEBUG, message, *args, **kwargs)
 
     def info(self, message, *args, **kwargs):
-        msg = self.createMessage(message, *args, **kwargs)
-        msg["level"] = "info"
-        self.logger.info(json.dumps(msg))
+        """Logs the message plus metadata at level ``logging.INFO``.
+
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
+        self._log(logging.INFO, message, *args, **kwargs)
 
     def warn(self, message, *args, **kwargs):
-        msg = self.createMessage(message, *args, **kwargs)
-        msg["level"] = "warn"
-        self.logger.warning(json.dumps(msg))
+        """Logs the message plus metadata at level ``logging.WARN``.
+
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
+        self._log(logging.WARN, message, *args, **kwargs)
 
     def warning(self, message, *args, **kwargs):
+        """Logs the message plus metadata at level ``logging.WARN``.
+
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
         self.warn(message, *args, **kwargs)
 
     def error(self, message, *args, **kwargs):
-        msg = self.createMessage(message, *args, **kwargs)
-        msg["level"] = "error"
-        self.logger.error(json.dumps(msg))
+        """Logs the message plus metadata at level ``logging.ERROR``.
+
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
+        self._log(logging.ERROR, message, *args, **kwargs)
 
     def fatal(self, message, *args, **kwargs):
-        msg = self.createMessage(message, *args, **kwargs)
-        msg["level"] = "fatal"
-        self.logger.error(json.dumps(msg))
+        """Logs the message plus metadata at level ``logging.CRITICAL``.
 
-    def trace(self, message, *args, **kwargs):
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
+        self.critical(message, *args, **kwargs)
+
+    def critical(self, message, *args, **kwargs):
+        """Logs the message plus metadata at level ``logging.CRITICAL``.
+
+        Passes the supplied arguments to ``createMessage`` and logs the
+        returned message dict.  See ``createMessage``.
+        """
+        self._log(logging.CRITICAL, message, *args, **kwargs)
+
+    def _resolve_log_level(self, level_name):
+        if not isinstance(level_name, str):
+            return self.logger.level
+        level = logging.getLevelName(level_name.upper())
+        return level if isinstance(level, (int, long)) else self.logger.level
```

Comment:
We need to support both python27 and python36, this works for Python 2 only.
Use one of these:
```python
from builtins import int    # subclass of long on Py2
if isinstance(x, int)

from past.builtins import long
if isinstance(x, (int, long))
```

Reply:
I ran this locally in Python 2.7 and the CI tests for 2.7 also passed. Did you see this fails somehow?

Reply:
It causes error in python 3.6.
pylint for python3 has: Undefined variable: 'long'
lambda with python 3.6 has: name 'long' is not defined: NameError

Reply:
I don't know why the tests passed in python3.6, that's something we want to check.

Reply:
Yeah, odd. I created a Python 3.6 env to run the tests and I'm able to reproduce the error. I forgot that Python 3 dropped `long`. I'll make the correction.